py-polars/polars/dataframe/frame.py

```diff
@@ -2091,7 +2091,7 @@ def to_numpy(
                 a = s.to_numpy()
                 arrays.append(
                     a.astype(str, copy=False)
-                    if tp == Utf8 and not s.has_validity()
+                    if tp == Utf8 and not s.null_count()
```

Comment:
I prefer this way of writing - this will indicate the method always returns an integer:

```suggestion
                    if tp == Utf8 and s.null_count() == 0
```

Same for the two other spots.

Reply:
That's what type-hinting is for: `def null_count(self) -> int:`. The additional `==` is unnecessary overhead (trivial, but unnecessary ;)

Reply:
All right, I'll let this one slide :smile: I'm always wary of using Python's implicit truthiness.